src/types.ts

```diff
@@ -16,7 +16,7 @@ interface ICache {
    * @returns
    * 'ok' if successful, else returns an Error object
    */
-  put: (key: string, data: any) => Promise<'ok' | Error>
+  put: (key: string, data: IResponse) => Promise<'ok' | Error>
```

Comment:
I think a good rule of thumb at this point is: No `IResponse` anywhere other than for `Response` constructor params.

```suggestion
  put: (key: string, data: Response) => Promise<'ok' | Error>
```